okhttp/third_party/okhttp/java/io/grpc/okhttp/internal/Platform.java

```diff
@@ -199,6 +209,23 @@ private static Platform findPlatform() {
       throw new RuntimeException(nsae);
     }
 
+    // Find JDK9+ ALPN support
+    try {
+      Method setApplicationProtocols =
+          SSLParameters.class.getMethod("setApplicationProtocols", String[].class);
+      Method getApplicationProtocol =
+          AccessController.doPrivileged(
+              new PrivilegedExceptionAction<Method>() {
+                @Override
+                public Method run() throws Exception {
+                  return SSLSocket.class.getMethod("getApplicationProtocol");
+                }
+              });
+      return new JdkAlpnPlatform(sslProvider, setApplicationProtocols, getApplicationProtocol);
```

Comment:
We should verify that `sslProvider` supports ALPN by calling `getApplicationProtocol()`.

Reply:
Done. AFAICT this requires instantiating an SSLEngine to invoke `getApplicationProtocol()`, which is the approach now taken.